doc/source/whatsnew/v0.21.0.txt

```diff
@@ -130,6 +130,8 @@ Numeric
 Categorical
 ^^^^^^^^^^^
 
+- Bug in ``Categorical.is_dtype_equal()`` where comparison with a Series with dtype='category' (:issue:`16659`)
```

Comment:
- Bug in ``Categorical.is_dtype_equal()`` where comparison with a Series with `'category'` dtype incorrectly returned False (:issue:`16659`)